pkg/service/metadata/drone/confd_server.go

```diff
@@ -5,39 +5,155 @@
 package drone
 
 import (
+	"encoding/json"
+	"fmt"
 	"sync"
 
-	"openpitrix.io/libconfd"
+	"github.com/gogo/protobuf/proto"
+
+	"openpitrix.io/openpitrix/pkg/libconfd"
+	"openpitrix.io/openpitrix/pkg/libconfd/backends"
+	"openpitrix.io/openpitrix/pkg/pb/types"
 )
 
 type ConfdServer struct {
-	confdProcessor *libconfd.Processor
-	confdCall      *libconfd.Call
-	confdMutex     sync.Mutex
+	mu            sync.Mutex
+	cfg           *pbtypes.ConfdConfig
+	config        *libconfd.Config
+	backendConfig *libconfd.BackendConfig
+
+	processor *libconfd.Processor
+	client    libconfd.BackendClient
+	running   bool
+	err       error
 }
 
 func NewConfdServer() *ConfdServer {
-	return &ConfdServer{
-		confdProcessor: libconfd.NewProcessor(),
+	return &ConfdServer{}
+}
+
+func (p *ConfdServer) IsRunning() bool {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	return p.processor != nil && p.running
+}
+
+func (p *ConfdServer) GetConfig() *pbtypes.ConfdConfig {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	if p.cfg != nil {
+		return p.cfg
 	}
+
+	return &pbtypes.ConfdConfig{}
 }
 
-func (p *ConfdServer) GetConfdConfig() *libconfd.Config {
-	return &libconfd.Config{}
+func (p *ConfdServer) SetConfig(cfg *pbtypes.ConfdConfig) error {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	config, backendConfig, err := p.parseConfig(cfg)
+	if err != nil {
+		return err
+	}
+
+	p.cfg = proto.Clone(cfg).(*pbtypes.ConfdConfig)
+	p.config = config
+	p.backendConfig = backendConfig
+
+	return nil
 }
 
-func (p *ConfdServer) GetBackendConfig() *libconfd.BackendConfig {
-	return &libconfd.BackendConfig{}
+func (p *ConfdServer) GetBackendClient() libconfd.BackendClient {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	return p.client
 }
 
-func (p *ConfdServer) Start(cfg *libconfd.Config, cfgBcakend *libconfd.BackendConfig) error {
+func (p *ConfdServer) Start(opts ...libconfd.Options) error {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	if p.processor == nil {
+		p.processor = libconfd.NewProcessor()
+	}
+	if p.running {
+		return fmt.Errorf("drone: confd is running!")
```

Comment:
Error string should not be capitalized or end with punctuation. 
Ref: https://github.com/golang/go/wiki/CodeReviewComments#error-strings